netmiko/utilities.py

```diff
@@ -344,6 +352,31 @@ def get_structured_data(raw_output, platform=None, command=None, template=None):
         )
 
 
+def get_structured_data_ttp(raw_output, template=None):
+    """
+    Convert raw CLI output to structured data using TTP template.
+
+    You can use a straight TextFSM file i.e. specify "template"
+    """
+    if not TTP_INSTALLED:
+        msg = "\nTTP is not installed. Please PIP install ttp:\n" "pip install ttp\n"
+        raise ValueError(msg)
+
+    if os.path.exists(template):
+        with open(template) as f:
+            ttp_template = f.read()
+        try:
+            if template:
+                ttp_parser = ttp(data=raw_output, template=ttp_template)
+                ttp_parser.parse()
+                # print(ttp_parser.result(format="json")[0])
+                return ttp_parser.result(format="raw")
```

Comment:
result methods support `structure` keyword, that is very useful to produce overall results combining them in different structure such as list of dictionaries, dictionary of flattening the lists. Moreover, result method support all the same attributes as outputters, all the way to calling functions and returners on results. suggest to introduce `ttp_res_kwarg` argument in addition to `template` keyword, `ttp_res_kwarg` can be a dictionary of attributes to unpack to result method such as:
```
return ttp_parser.result(**ttp_res_kwarg)
```

Reply:
Sure, nw. 

One more thing, please pay attention to `ttp_parser.parse(one=True) # force to parse in single process` line and `one=True` in particular, as TTP can invoke multiprocessing mode if data to parse exceeds 5 MByte in size, that might not work out of the box when TTP used as a module, hence `one=True` a safer way of running.